libs/router/src/lib/route.component.ts

```diff
@@ -59,12 +51,16 @@ interface State {
   providers: [
     {
       provide: RouteParams,
-      useFactory: getRouteParams,
+      useFactory(routeComponent: RouteComponent) {
+        return routeComponent.routeParams$;
+      },
       deps: [[new Self(), RouteComponent]],
```

Comment:
nit: why not go full `inject()` 😛
```suggestion
      useFactory() {
        return inject(RouteComponent, InjectFlags.Self).routeParams$;
      }
```